tutorials/send-email-with-sparkpost-and-nodejs-on-google-app-engine.md

```diff
@@ -0,0 +1,158 @@
+
+    ```sh
+    npm install --save express body-parser pug sparkpost
+    ```
+
+## Create
+
+1. Create an `server.js` file with the following contents:
+
+    ```js
+    'use strict';
+
+    const http = require('http');
+    const express = require('express');
+    const path = require('path');
+    const bodyParser = require('body-parser');
+
+    const SparkPost = require('sparkpost');
+    const spClient = new SparkPost(process.env.SPARKPOST_API_KEY);
+
+    const app = express();
+    const srv = http.Server(app);
+
+    // Setup view engine
+    app.set('views', path.join(__dirname, 'views'));
+    app.set('view engine', 'pug');
+
+    // Parse form data
+    app.use(bodyParser.json());
+    app.use(bodyParser.urlencoded({ extended: false }));
+
+    app.get('/', (req, res) => res.render('index'));
+
+    app.post('/hello', (req, res, next) => {
+      spClient.transmissions.send({
+        options: { sandbox: true },
+        content: {
+          from: 'appengine-node-demo@sparkpostbox.com',
+          subject: 'Hello from Google AppEngine!',
+          html:'<html><body><p>Google AppEngine + Node.js + SparkPost = awesome!</p></body></html>'
+        },
+        recipients: [
+          {address: req.body.email} 
+        ]
+      }).then(result => {
+        res.render('index', {sent: true});
+      }).catch(err => {
+        res.render('index', {err: err});
+        console.error(err);
+      });
+    });
+
+    srv.listen(process.env.PORT || 8080, () => {
+      console.log(`Listening on ${srv.address().port}`);
+    });
+
+    ```
+
+1. Create a directory named `views`:
+
+    ```sh
+    mkdir views
+    ```
+
+1. Create a file named `index.pug` inside the `views` directory with the
+following contents:
+
+    ```
+    doctype html
+    html    
+      head
+        title= title
+      body
+        h1 hello world!
+        p express.js + sparkpost on google app engine.
+        hr
+        if sent
+          p email sent!
+        if err
+          p Oh my. Something's not right:
+            ul
+              each e in err.errors
+                li
+                  strong=e.message+': '
+                  |#{e.description}
+        else
+          form(name="hello", action="/hello", method="post")
+            input(type="email", placeholder="enter your email to send yourself a hello world message", name="email", style="width: 50%; margin-right: 15px;")
+            input(type="submit", value="send")
+    ```
+
+## Run
+
+1. Run the app with the following command:
+
+    ```sh
+    SPARKPOST_API_KEY=your-sparkpost-api-key npm start
+    ```
+
+1. Visit [http://localhost:8080](http://localhost:8080) to try sending an email.
+
+## Deploy
+
+    ```yaml
+    runtime: nodejs
+    env: flex
+    env_variables:
+      SPARKPOST_API_KEY: your-sparkpost-api-key
+    ```
+
+    The `app.yaml` makes the app deployable to Google App Engine Managed VMs.
```

Comment:
I think the indent in this line is causing the yaml above not to render correctly in the preview view.